sys/kern/proc.c

```diff
@@ -152,30 +178,39 @@ void proc_add(proc_t *p) {
   WITH_MTX_LOCK (all_proc_mtx) {
     p->p_pid = pid_alloc();
     TAILQ_INSERT_TAIL(&proc_list, p, p_all);
+    TAILQ_INSERT_TAIL(PROC_HASH_CHAIN(p->p_pid), p, p_hash);
     if (p->p_parent)
       TAILQ_INSERT_TAIL(CHILDREN(p->p_parent), p, p_child);
   }
 
   klog("Process PID(%d) {%p} has been created", p->p_pid, p);
 }
 
-proc_t *proc_find(pid_t pid) {
+/* Lookup a process in the PID hash table.
+ * The returned process, if any, is NOT locked. */
+proc_t *proc_find_raw(pid_t pid) {
```

Comment:
You defined it `static` earlier? How come the symbol leaks outside?!

Reply:
An omission on my part, the function is meant to be static.